quantuminspire/cli/command_list.py

```diff
@@ -237,9 +239,26 @@ def upload_files(name: str = typer.Argument(..., help="The name of the file to u
     runtime = RemoteRuntime()
     program = HybridAlgorithm(platform_name="spin-2", program_name=name)
     program.read_file(Path(name))
-    batch_run = runtime.run(program)
+    run_id = runtime.run(program)
     typer.echo(f"Upload file with name: {name}")
-    typer.echo(f"batch_run_id {batch_run.id} with runs f{batch_run.run_ids}")
```

Comment:
I think this change is a regression. The id returned with this new change seems to be the id of the batch run and print it as "run_id", this is very confusing. As you can see on https://qutech-sd.gitlab.io/quantum-inspire-2/platform/latest/design/code/models/compute_job_manager/algorithms.html a batch run consists of multiple runs which each have their own id. And result and FinalResult belong to a run, and not a batchrun.  And we should retrieve the results of one of the ids of the run_ids not of the batch_run id since that doesn't have any results attached to them.

Reply:
It is confusing, but it's a result of the runtime classes setup we currently have. The batch run concept doesn't exist outside the remote runtime, but the base class needs to define the return type for the run method.